### PR TITLE
Add net.tedomum.CapBattleship

### PR DESCRIPTION
**Pull request by linkmauve**
This is a battleship game written in Python using pygame.

The metainfo file, setup.py patch and complicated build commands are all already upstreamed in master, a new alpha5 version is pending the inclusion of a SVG icon.

### CODE REVIEW

**Comment by linkmauve** [comment]
bot, build net.tedomum.CapBattleship

**Comment by linkmauve** [comment]
This game currently hardcodes paths in /usr, this will have to be fixed upstream before this flatpak can continue.

**Comment by linkmauve** [comment]
bot, build net.tedomum.CapBattleship

**Comment by linkmauve** [comment]
bot, build net.tedomum.CapBattleship

File: net.tedomum.CapBattleship.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    "runtime-version": "20.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "capbattleship",
+    "finish-args": [
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--socket=pulseaudio",
+        "--device=dri"
+    ],
+    "cleanup": [],
```
Comment (bilelmoussaoui) [inline net.tedomum.CapBattleship.json:13]:
```suggestion
```

File: net.tedomum.CapBattleship.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            "name": "capbattleship",
+            "buildsystem": "simple",
+            "build-commands": [
+                "pip3 install --prefix=/app .",
+                "mkdir -p /app/share/icons/hicolor/256x256/apps/",
+                "mkdir -p /app/share/applications/",
+                "install -Dm644 net.tedomum.CapBattleship.metainfo.xml /app/share/metainfo/net.tedomum.CapBattleship.metainfo.xml",
+                "mv /app/share/pixmaps/capbattleship.png /app/share/icons/hicolor/256x256/apps/net.tedomum.CapBattleship.png",
+                "sed '/^Icon/s/=.*/=net.tedomum.CapBattleship/' /app/share/applications/capbattleship.desktop > /app/share/applications/net.tedomum.CapBattleship.desktop",
```
Comment (bilelmoussaoui) [inline net.tedomum.CapBattleship.json:42]: Should be renamed upstream otherwise use desktop-file-edit

File: net.tedomum.CapBattleship.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        },
+        {
+            "name": "capbattleship",
+            "buildsystem": "simple",
+            "build-commands": [
+                "pip3 install --prefix=/app .",
+                "mkdir -p /app/share/icons/hicolor/256x256/apps/",
+                "mkdir -p /app/share/applications/",
+                "install -Dm644 net.tedomum.CapBattleship.metainfo.xml /app/share/metainfo/net.tedomum.CapBattleship.metainfo.xml",
+                "mv /app/share/pixmaps/capbattleship.png /app/share/icons/hicolor/256x256/apps/net.tedomum.CapBattleship.png",
```
Comment (bilelmoussaoui) [inline net.tedomum.CapBattleship.json:41]: install would work fine here as well

File: net.tedomum.CapBattleship.json (hunk 2 above)
Comment (linkmauve) [reply net.tedomum.CapBattleship.json:42]: This is already fixed upstream, I’m just waiting for a release from them.

File: net.tedomum.CapBattleship.json (hunk 3 above)
Comment (linkmauve) [reply net.tedomum.CapBattleship.json:41]: Already fixed upstream too.

**Comment by linkmauve** [comment]
There!  alpha5 is released, this PR updated, everything merged upstream! :)

**Comment by linkmauve** [comment]
bot, build net.tedomum.CapBattleship

**Comment by barthalion** [comment]
Have you asked upstream developer if they would like to maintain their app on Flathub?

**Comment by captnfab** [comment]
Hi @barthalion , I'm upstream author, and I'm perfectly fine with @linkmauve maintaining the app on Flathub :-)
Though, it's probably a good thing if I can maintain it too in case of problems.

**Comment by barthalion** [comment]
Great! Exactly why I asked – we want to make sure the author can do any changes should they want to. :+1: 

/merge @captnfab

**Comment by captnfab** [comment]
Thanks!

**Pull request closed without merging**